src/vnm_mad/remotes/802.1Q/vlan_tag_driver.rb

```diff
@@ -66,21 +68,23 @@ def create_vlan_dev
 
         # Do not fail if the device exists to prevent race conditions.
         # ip link add returns 2 on "RTNETLINK answers: File exists"
-        OpenNebula.exec_and_log("#{command(:ip)} link add link"\
+        LocalCommand.run_sh("#{command(:ip)} link add link"\
             " #{@nic[:phydev]} name #{@nic[:vlan_dev]} #{mtu} type vlan id"\
-            " #{@nic[:vlan_id]} #{ip_link_conf}", nil, 2)
+            " #{@nic[:vlan_id]} #{ip_link_conf}", :ok_rcs => 2)
 
-        OpenNebula.exec_and_log("#{command(:ip)} link set #{@nic[:vlan_dev]} up")
+        LocalCommand.run_sh("#{command(:ip)} link set #{@nic[:vlan_dev]} up")
     end
 
     def delete_vlan_dev
-        OpenNebula.exec_and_log("#{command(:ip)} link delete"\
+        LocalCommand.run_sh("#{command(:ip)} link delete"\
             " #{@nic[:vlan_dev]}") if @nic[:vlan_dev] != @nic[:phydev]
     end
 
+    # rubocop:disable Style/CommandLiteral
+    # rubocop:disable Style/SpecialGlobalVars
     def list_interface_vlan(name)
-        text = %x(#{command(:ip_unpriv)} -d link show #{name})
-        return nil if $?.exitstatus != 0
+        text = `#{command(:ip_unpriv)} -d link show #{name}`
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Detected a dangerous subshell execution vulnerability where user input is directly used in a shell command without proper sanitisation. This could allow an attacker to inject and execute arbitrary system commands, potentially leading to system compromise or data breach. To remediate, avoid using shell commands with user input when possible, or implement strict input validation and use shell escaping functions like Shellwords.escape(). If shell execution is necessary, consider using a allow list of allowed commands and arguments. Link to more info : https://guides.rubyonrails.org/security.html